packages/core/src/ruleset/rule.ts

```diff
@@ -24,9 +24,10 @@ export interface IRule {
   documentationUrl: string | null;
   then: IRuleThen[];
   given: string[];
+  extensions: Record<string, unknown> | null;
 }
 
-type RuleJson = Omit<IRule, 'then'> & {
+type RuleJson = Omit<IRule, 'then' | 'extensions'> & {
```

Comment:
```suggestion
type RuleJson = Omit<IRule, 'then'> & {
```
Let's have it included :)

Reply:
> Looks good to me! One minor request - could you please add `extensions` to
> 
> https://github.com/stoplightio/spectral/blob/362e74f5b371507a5e3124e6c74afc0061ca071f/packages/core/src/ruleset/rule.ts#L172
> 
> ?
> ```diff
>        resolved: this.resolved,
>        formats: this.formats,
> +      extensions: this.extensions,
>        then: this.then.map(then => ({
> ```

As extensions type is `Record<string, unknown>` we don't know what comes as a value here, it's `unknown`... it might be not a valid "JSON" in this sense.

Alternatively, we can allow only primitives as values in this object to make sure it's a JSON compatible object.

Reply:
> it's unknown... it might be not a valid "JSON" in this sense

Well, it's going to be JSON for JSON rulesets, but yeah, could be a non-serializable value in the case of JS rulesets.

> Alternatively, we can allow only primitives as values in this object to make sure it's a JSON compatible object.

objects and arrays are both JSON compatible objects too.
I believe we can leave it as is - JSON.stringify will just drop any invalid values.

Reply:
done